test/unit/reductions/grid_search/test_grid_generator.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from fairlearn.reductions._grid_search.grid_search import _GridGenerator
+from fairlearn.reductions import DemographicParity, EqualizedOdds
+from fairlearn.reductions._moments.conditional_selection_rate import _ALL, _SIGN, _EVENT, \
+    _GROUP_ID
+
+from test.unit.reductions.grid_search import _utilities
+
+
+@pytest.mark.parametrize("grid_size", [5, 6, 10, 11, 100, 101, 102])
+@pytest.mark.parametrize("grid_limit", [1.0, 2.0, 4.0, 10.0])
+def test_grid_generator_demographic_parity(grid_size, grid_limit):
+    disparity_moment = DemographicParity()
+    events = [_ALL]
+
+    grid = calculate_grid(grid_limit, grid_size, disparity_moment, events)
+
+    expected_index = pd.MultiIndex.from_product([['+', '-'], events, [0, 1]],
+                                                names=[_SIGN, _EVENT, _GROUP_ID])
+    assert (expected_index == grid.index).all()
+    expected_grid = pd.DataFrame()
+    grid_size_or_next_smaller_even_number = int(grid_size / 2) * 2
```

Comment:
As I recall, this was one of the peculiarities of the GridGenerator I found when I was prodding at it - a grid size of 2 would give points {-1,0} and a size of 3 would give {-1,0,1}. Is that what this is about?

Reply:
Yes!